src/usr/diag/prdf/common/framework/register/prdfErrorRegister.C

```diff
@@ -120,8 +120,6 @@ ErrorRegister::ErrorRegister( SCAN_COMM_REGISTER_CLASS & r, ResolutionMap & rm,
     ErrorRegisterType(), scr(r), scr_rc(SUCCESS), rMap(rm),
     xNoErrorOnZeroScr(false), xScrId(scrId)
 {
-    PRDF_ASSERT( &r  != NULL );
```

Comment:
Hrm.... so dereferencing a null pointer is undefined behavior.

From the standard:
<blockquote>
A reference shall be initialized to refer to a valid object
or function. [
Note:
in particular, a null reference cannot exist in a well-defined program, because the only
way to create such a reference would be to bind it to the “object” obtained by dereferencing a null pointer,
which causes undefined behavior. As described in
9.6, a reference cannot be bound directly to a bit-field.
— end note
]
</blockquote>

So if there isn't a suitable check that `p != NULL` then I guess either the compiler or a static analysis tool should warn/error.... as it seems the check of the address of the reference would itself be undefined behavior for a null pointer.